mbed_targets/_internal/target_attributes.py

```diff
@@ -30,6 +32,17 @@ class TargetAttributesNotFoundError(TargetAttributesError):
     """Attributes for target not found in targets.json."""
 
 
+@dataclass
+class MbedTargetAttributes:
+    """A set of build attributes for an Mbed Target.
+
+    As defined in the mbed-os repository's targets.json config file.
```

Comment:
I'd prefer we had an `Attributes` section in this docstring describing what the attributes are. Even though this is a simple object it's still good practice to document the attributes.

Reply:
ed78110